ores/tests/test_api.py

```diff
@@ -1,6 +1,54 @@
 from .. import api
 
 
+class FakeSession:
+
+    def __init__(self, error_out=False):
+        self.error_out = error_out
+
+    def get(self, url, params, headers, verify=False, stream=False):
+        if self.error_out:
```

Comment:
This is just what MagicMock is for, you could define a mock to do this without extra state, in test_base_error below.

```
import mock
with mock.patch('requests.Session', return_value=FAKE_RESPONSE) as MockSession:
    api.Session("myhost")
```

Reply:
Not sure I see what is going on here.  How does the MockSession know how what to return when .get() is called.  Maybe I need to study up more :|

Reply:
Oh!  I guess we could monkey patch the `_session` variable in the test.  I feel conflicted about that though.

Reply:
It's just those mock lines I wrote above.  Either a mock or a fake object like you have is fine, but we should never include logic in production code which will only be used by tests.  A mock is also much simpler and easier to read because the special behavior is defined at the site of use.

To answer this question
> How does the MockSession know how what to return when .get() is called?

You would use the mock lines I pasted in order to provide an error from get(), or you can define the return_value as the happy-path function.  A mock also lets you assert that the method is called with expected arguments.